subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoTaskExtension.java

```diff
@@ -52,11 +54,12 @@ public String getAsArg() {
         }
     }
 
+    private Project project;
```

Comment:
In this case we really only need the `Project` instance to create a `Provider`. Are you suggesting that we allow creating providers differently or somehow make the project instance available automatically for an extension?